### PR TITLE
Add Lambda filter

### LINKED ISSUE #119: Lambda filter
We should have a Lambda filter Class to enable filter extension with custom functions. 

Inspired by https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.Lambda

### PR DESCRIPTION
**Pull request by alessiamarcolini**
Fix #119

### CODE REVIEW

File: src/histolab/filters/image_filters.py [hunk 1]
```diff
@@ -43,6 +43,34 @@ def __call__(self, img: PIL.Image.Image) -> Union[PIL.Image.Image, np.ndarray]:
...
+
+    Returns
+    -------
+    PIL.Image.Image
+        The image with the function applied.
+    """
+
+    def __init__(self, lambd: Callable[[PIL.Image.Image], PIL.Image.Image]) -> None:
+        assert callable(lambd), repr(type(lambd).__name__) + " object is not callable"
+        self.lambd = lambd
```
Comment (nicolebussola) [inline src/histolab/filters/image_filters.py:65]: maybe lmb istead of lambd?

Comment (alessiamarcolini) [reply src/histolab/filters/image_filters.py:65]: mmm I just copied it from torchvision, do you think lmb is better..?

Comment (nicolebussola) [reply src/histolab/filters/image_filters.py:65]: uh ok.... I just do not like lambda without the a.... but ok let's use torchvision style

Comment (alessiamarcolini) [reply src/histolab/filters/image_filters.py:65]: It's not possible to use lambda because it's a python keyword 🤷🏻‍♀️

**Pull request merged**